### PR TITLE
Restore spec compliant structure of MatchCriteria

### PR DESCRIPTION
**Pull request by johannesjasper**

### CODE REVIEW

File: fido-uaf-core/src/main/java/com/nexenio/fido/uaf/core/message/MatchCriteria.java [hunk 1]
```diff
@@ -32,4 +33,32 @@
     @JsonProperty("aaid")
     private String[] aaids;
 
+    @SerializedName("vendorID")
+    @JsonProperty("vendorID")
+    private List<String> vendorIDs;
```
Comment (Steppschuh) [inline fido-uaf-core/src/main/java/com/nexenio/fido/uaf/core/message/MatchCriteria.java]: Rename to `vendorIds` for consistency (like in `FinalChallengeParams.java`) and to meet [conventions](https://stackoverflow.com/questions/1699944/what-is-correct-java-naming-convention-for-id).

File: fido-uaf-core/src/main/java/com/nexenio/fido/uaf/core/message/MatchCriteria.java [hunk 2]
```diff
@@ -32,4 +33,32 @@
...
+
+    private List<String> keyIDs;
```
Comment (Steppschuh) [inline fido-uaf-core/src/main/java/com/nexenio/fido/uaf/core/message/MatchCriteria.java]: Rename to `keyIds`

**Review by Steppschuh — CHANGES_REQUESTED** [review]

**Review by Steppschuh — DISMISSED** [review]

File: fido-uaf-core/src/main/java/com/nexenio/fido/uaf/core/message/MatchCriteria.java [hunk 3]
```diff
@@ -34,4 +34,34 @@
...
+
+    private long attachmentHint;
+
+    private int tcDisplay;
+
+    private List<Integer> authenticationAlgorithms = new ArrayList<>();
+
+    private List<String> assertionSchemes = new ArrayList<>();
+
+    private List<Integer> attestationTypes = new ArrayList<>();
```
Comment (johannesjasper) [inline fido-uaf-core/src/main/java/com/nexenio/fido/uaf/core/message/MatchCriteria.java:59]: @Steppschuh  Should lists be `null` or empty?

File: fido-uaf-core/src/main/java/com/nexenio/fido/uaf/core/message/MatchCriteria.java [hunk 4]
```diff
@@ -34,4 +34,34 @@
...
+
+    private long userVerification;
+
+    private int keyProtection;
+
+    private int matcherProtection;
+
+    private long attachmentHint;
+
+    private int tcDisplay;
```
Comment (Steppschuh) [inline fido-uaf-core/src/main/java/com/nexenio/fido/uaf/core/message/MatchCriteria.java:53]: Before looking it up i didn't know that this stands for `transaction confirmation display`. Maybe we can rename that.

**Review by Steppschuh — APPROVED** [review]

**Pull request merged**